sphinx/environment.py

```diff
@@ -961,6 +962,29 @@ def process_downloads(self, docname, doctree):
 
     def process_images(self, docname, doctree):
         """Process and rewrite image URIs."""
+        def collect_candidates(imgpath, candidates):
+            globbed = {}
+            for filename in glob(imgpath):
+                new_imgpath = relative_path(path.join(self.srcdir, 'dummy'),
+                                            filename)
+                if filename.lower().endswith('.pdf'):
+                    if 'application/pdf' not in candidates:
+                        globbed.setdefault('application/pdf', []).append(new_imgpath)
```

Comment:
I might be misunderstanding the logic here, but you're trying to stop collecting further candidates for a mimetype if we already have one?  I'd suggest doing that in the loop further below where you actually fill up `candidates`, rather than sprinkling it over here.